### PR TITLE
Techdocs cleanup for local

### PR DESCRIPTION
**Pull request by ebarriosjr**
## Hey, I just made a Pull Request!

Currently techdocs will copy /tmp/backstage-repo into every docs tmp folder it generates. If you have multiple services the size of this folder can grow and grow. If the git repos cloned into /tmp/backstage-repo are big then your container size will explode.

This PR removes the nackstage-repo from the generated docs tmp folder.
This PR also also converts the copy of the resulting docs from /tmp to a move to clean up the /tmp folder in the container.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by OrkoHunter** [comment]
Hey @ebarriosjr ! Could you tell what version of `techdocs-backend` are you using? 

> Currently techdocs will copy /tmp/backstage-repo into every docs tmp folder it generates.

I'm sorry but I don't see this when I look at the contents of the generated folder. I also see no code which does this copy. Please correct me if I am wrong.

<details>
<summary>(Expand me) Example generated directory content</summary>

There is no `backstage-repo` directory here.

```
▶ cd /private/var/folders/pw/37gxzy591bs8yvw1hptwksxr0000gn/T/techdocs-tmp-JckWIc

37gxzy591bs8yvw1hptwksxr0000gn/T/techdocs-tmp-JckWIc
▶ ls
404.html                          openapi
FAQ                               overview
api                               plugins
architecture-decisions            prettier.config.js
assets                            reference
auth                              search
cli                               service_specification.schema.json
conf                              sitemap.xml
dls                               sitemap.xml.gz
features                          support
generate-uml.sh                   techdocs_metadata.json
getting-started                   tutorials
index.html
```

</details>

To re-iterate the flow as I understand (in local publisher) -

1. In prepare step, TechDocs clones the repository at `</path/to/tmp>/backstage-repo/<host|github.com/org/repo/branch>`
2. For future updates, techdocs does a `git pull` here.
3. In generate step, TechDocs runs `mkdocs build` on this(1) directory and outputs contents at a new directory `</path/to/tmp>/techdocs-tmp-<abc123>/`
4. The source repo is not copied here, as I see it. So the `rm -rf </path/to/tmp>/techdocs-tmp-<abc123>/backstage-repo` you are suggesting may not be useful.
5. In publish step, TechDocs copies the content of this(3) generated directory to `static` folder of the path where `techdocs-backend` plugin lives.

File: packages/techdocs-common/src/stages/publish/local.ts [hunk 1]
```diff
@@ -84,6 +84,7 @@ export class LocalPublish implements PublisherBase {
           .catch(reason => {
             reject(reason);
           });
+        fs.remove(directory);
```
Comment (OrkoHunter) [inline packages/techdocs-common/src/stages/publish/local.ts:87]: This is indeed useful. We don't really need the generated directory once it has been published to a more non-temporary location.

**Comment by ebarriosjr** [comment]
Hi @OrkoHunter my techdocs-backend version is:
@backstage/plugin-techdocs-backend": "^0.5.0"

**Comment by OrkoHunter** [comment]
Aah, that's recent enough. I wonder why is `backstage-repo` being copied over in all of your generated folders. :( I'll appreciate if you can help debug.

**Comment by freben** [comment]
Hey! Any progress on this? Seems valuable!

**Comment by OrkoHunter** [comment]
Yep! `fs.remove` bit is valuable indeed. No need to keep the "generated" directory around, once it has been published.

I am not convinced about `rm -rf` still, since it is an unexpected behavior that a "prepared" directory is being copied over to a "published" directory.

But not denying that the bug may not exist. Investigating on https://github.com/backstage/backstage/issues/4011 and there might be something else going on.

For now let's move ahead with only `fs.remove`.

**Comment by freben** [comment]
Ok. The original poster has been away for a while it seems - could be considered for takeover if you have the bandwidth.

Just going through the PR history to see if there's stuff to move ahead / ping / close :)

**Comment by OrkoHunter** [comment]
Yep, I can take this. 💯 

And thank you for the backlog cleanup. Was much needed TBH :)

**Comment by freben** [comment]
Rudely assigned just to mark for posterity :)

**Comment by freben** [comment]
(and a small ping)

**Comment by OrkoHunter** [comment]
Yep thanks! I will close this now, and re-open a new PR this week - Already started on it!

**Pull request closed without merging**